cpp/src/arrow/array/data.h

```diff
@@ -434,6 +434,14 @@ struct ARROW_EXPORT ArraySpan {
     return GetValues<T>(i, this->offset);
   }
 
+  // Access a buffer's data as a span
+  template <typename T>
+  util::span<const T> GetSpan(int i) const {
+    return util::span(buffers[i].data_as<T>(),
+                      static_cast<size_t>(buffers[i].size) / sizeof(T))
+        .subspan(offset);
+  }
+
```

Comment:
we should probably also include an overload returning a mutable span
```suggestion

  template <typename T>
  util::span<T> GetSpan(int i) {
    return util::span(buffers[i].data_as<T>(),
                      static_cast<size_t>(buffers[i].size) / sizeof(T))
        .subspan(offset);
  }
```

Reply:
Done